### PR TITLE
JPERF-1043 Add support for list view on IssueNavigatorPage. Add Issue…

### PR DESCRIPTION
**Pull request by ewefie**
Add support for list view on IssueNavigatorPage:
- Add IssueNavigatorView enum class that represents both available layouts and use it as an optional parameter in SearchJqlSimpleAction and SearchJqlSimpleAction constructors. When no value is provided, the detail view will be used as a default layout.
- While visiting the IssueNavigatorPage current view is detected and changed to the desired view if needed.

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt
Comment (ewefie) [inline src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: Unfortunately, to detect the currently selected view we need to wait for search results to appear.

Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: Hmm, we're using the sequence of `switchLayoutTo` and then `waitForIssueNavigator`. So if `By.className("results-panel")` is earlier than the `waitForIssueNavigator` conditions, it will not affect the measurements. In other words, if `waitForIssueNavigator` was already waiting for `results-panel` (indirectly), then wer're good.

If not, then we could do the switch *after* the page load. It would distort the proportions of the layouts, but it would preserve the measurement, which is more important.

Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: So I checked on dogfooding and indeed `results-panel` is a container for all the selectors in `val detailView` and `val listView`. So we're good.

But, the sequence is  `goToIssueNavigator(jqlQuery).switchLayoutTo(view).waitForIssueNavigator()`. So the `goToIssueNavigator` already performs the search, then `switchLayoutTo` can interrupt it, switch layout and re-trigger the search. It means that sometimes we'll make 1,5 search, affecting measurements.

So we can:
* bite the bullet and count on a low switch count
* switch after search
* prevent early search (no `jql=` in URI)

The `goToIssueNavigator` triggers JQL, because historically there was no other way. Going to "search for issues" was sending an empty JQL, which was interpreted as "all issues".
image
Since then, Jira changed empty JQL to mean "no search". So technically we can now `goToIssueNavigator` without a JQL, switch layout and then search.

**Review by dagguh — CHANGES_REQUESTED** [review]
Good stuff overall, but let's discuss the measurement implications

File: src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt
Comment (ewefie) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: I chose this order of operations because, as you have noticed, to detect the currently selected layout, we need the `results-panel` to be present. If we reverse the order, there are some questions that arise: 
Should changing the layout be included in the measurement? If so, shouldn't we repeat the verification which is currently being done in the `waitForIssueNavigator` function at the end? If not, do we even need to change the layout at all? 
I must admit that I didn't consider the option of separating the search from actually going to the Issue Search view, perhaps that option is the best...

**Comment by pczuj** [comment]
I don't have anything more to add to the review. Waiting with approve for application of suggestions.

File: src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt
Comment (ewefie) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:47]: Sure, it was a bit hard to pick one as both names are used in Jira, i.e.: we have "split-view" (sometimes called "detail-view") and "list-view" but to switch between them you have to click on `layout-switcher-button` an choose the option with proper `data-layout-key` 🙄 .

**Comment by dagguh** [comment]
Feedback addressed in #118

File: src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: Switching views should be measured separately from searching issues. We can optimistically assume that switches won't be that frequent, if the scenario doesn't mix both views in one action list. But even then we can expect a lot of switches when the same Jira user is reused across VUs.

So the optimistic assumption must come with a verification - counting the switches. I'll just make a nested action for the switching part. It will have to inform us about each switch attempt and whether it actually switched or just detected that the view is already there. I'll do that in my PR->PR (#118).

Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueNavigatorPage.kt:55]: Done in https://github.com/atlassian/jira-actions/pull/118/commits/0acb5fc123efd40c4e1b270e40e56958c8c1f9dc

**Comment by dagguh** [comment]
Red CI, investigating...

File: src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/action/SearchIssues.kt
Comment (pczuj) [inline src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/action/SearchIssues.kt:100]: If this is a private entity under scope of `SearchIssues` (it's in the `SearchIssues.kt` file), shouldn't it be inside the `SearchIssues` class?

**Comment by dagguh** [comment]
Pushed in PR fixes and a rebase onto #120. After #120 merge, the commit list should be shorter.

**Comment by dagguh** [comment]
All PR feedback is addressed (resolved conversations). We just wait for #120, rebase, green CI.

**Review by pczuj — APPROVED** [review]

**Comment by dagguh** [comment]
#120 merged, this rebased onto master, awaiting CI...

**Comment by pczuj** [comment]
#120 is merged

**Comment by dagguh** [comment]
CI doesn't propagate to checks, another [incident](https://www.githubstatus.com/incidents/z3c6q056q332) :grimacing:

**Pull request merged**
